src/components/FilenameInput.jsx

```diff
@@ -49,7 +51,7 @@ export default class FilenameInput extends Component {
 
   submit () {
     this.setState({ working: true, error: false })
-    this.props.onSubmit(this.state.value)
+    this.props.onSubmit && this.props.onSubmit(this.state.value)
```

Comment:
We shall use [PropTypes](https://facebook.github.io/react/docs/typechecking-with-proptypes.html#proptypes) and its `isRequired` API instead of this code.